source/DataWriter.h

```diff
@@ -96,23 +124,40 @@ class DataWriter {
 // The Write() function can take any number of arguments, each of which becomes
 // a token. They must be either strings or numeric types.
 template <class A, class ...B>
-void DataWriter::Write(const A &a, B... others)
+DataWriter &DataWriter::Write(const A &a, B... others)
 {
 	WriteToken(a);
 	Write(others...);
+	return *this;
 }
 
 
 
 // Write any numeric type as a single token.
 template <class A>
-void DataWriter::WriteToken(const A &a)
+DataWriter &DataWriter::WriteToken(const A &a)
 {
 	static_assert(std::is_arithmetic<A>::value,
 		"DataWriter cannot output anything but strings and arithmetic types.");
 
-	out << *before << a;
-	before = &space;
+	WriteSeparator();
+	out << a;
+	before = &separator;
+	return *this;
+}
+
+#include <iostream>
+
+// Writes a series of tokens without terminating the line.
+template <class A, class ...B>
+DataWriter &DataWriter::WriteToken(const A &a, B... others)
+{
+	std::cerr << "token" << std::endl;
+	WriteToken(a);
+	const std::size_t n = sizeof...(B);
+	if(n)
```

Comment:
You don't need this, because it will call the WriteToken function from above if only one argument is passed.